### PR TITLE
Feature/yext

### PR DESCRIPTION
**Pull request by AlexTeusz**
This Custom Module interacts with the [Yext Developer API](https://developer.yext.com/docs/api-reference/#tag/Knowledge-Manager) and provides 4 custom nodes: 

- GetEntity
- GetEntityById
- GetLocationsByFilter
- CreateLocation

Read more about Yext [here](https://www.yext.de/)

### CODE REVIEW

**Comment by robinschuster** [comment]
I cannot use the getlocationsbyfilter node. Even if i fill out all fields as described, i get "error": "Request failed with status code 400" in the context

**Comment by robinschuster** [comment]
I am having the same issues with the createLocation node

File: modules/yext/src/module.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+ * @arg {CognigyScript} `locationName` The name of the new location
+ * @arg {CognigyScript} `address` The address of the new location
+ * @arg {CognigyScript} `city` The city of the new location
+ * @arg {CognigyScript} `state` The state of the new location
+ * @arg {CognigyScript} `zip` The zip code of the new location
+ * @arg {CognigyScript} `countryCode` The country code of the new location
+ * @arg {CognigyScript} `phone` The phone of the new location
+ * @arg {CognigyScriptArray} `categoryIds` The categoryIds for the new location
+ * @arg {CognigyScript} `featuredMessage` The featuredMessage of the new location
+ * @arg {CognigyScript} `api_version` The version you want to use, e.g. 20190424 (a date)
```
Comment (robinschuster) [inline modules/yext/src/module.ts]: i honestly don't know what to put in here

File: modules/yext/src/module.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            city: args.city,
+            state: args.state,
+            zip: args.zip,
+            countryCode: args.countryCode,
+            phone: args.phone,
+            categoryIds: args.categoryIds,
+            featuredMessage: args.featuredMessage
+        }
+
+        axios.post(`https://api.yext.com/v2/accounts/me/locations?api_key=${args.secret.api_key}&v=${version}`, data, {
```
Comment (robinschuster) [inline modules/yext/src/module.ts]: please use the `params` option from axios for submitting query params

File: modules/yext/src/module.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    // Check if secret exists and contains correct parameters
+    if (!args.secret || !args.secret.api_key) return Promise.reject("Secret not defined or invalid.");
+    if (!args.entity) return Promise.reject("No entity defined.");
+
+    let version = args.api_version || "20190424"
+
+    return new Promise((resolve, reject) => {
+        let result = {};
+        
+        axios.get(`https://api.yext.com/v2/accounts/me/${args.entity.toLowerCase()}?api_key=${args.secret.api_key}&v=${version}`, {
```
Comment (robinschuster) [inline modules/yext/src/module.ts]: please use the `params` option from axios for submitting query params

File: modules/yext/src/module.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    // Check if secret exists and contains correct parameters
+    if (!args.secret || !args.secret.api_key) return Promise.reject("Secret not defined or invalid.");
+    if (!args.entity) return Promise.reject("No entity defined.");
+
+    let version = args.api_version || "20190424"
+
+    return new Promise((resolve, reject) => {
+        let result = {};
+        
+        axios.get(`https://api.yext.com/v2/accounts/me/${args.entity.toLowerCase()}/${args.entity_id}?api_key=${args.secret.api_key}&v=${version}`, {
```
Comment (robinschuster) [inline modules/yext/src/module.ts]: please use the `params` option from axios for submitting query params

File: modules/yext/src/module.ts [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+
+    // Check if secret exists and contains correct parameters
+    if (!args.secret || !args.secret.api_key) return Promise.reject("Secret not defined or invalid.");
+
+    let version = args.api_version || "20190424"
+
+    return new Promise((resolve, reject) => {
+        let result = {};
+        
+        axios.get(`https://api.yext.com/v2/accounts/me/locationsearch?api_key=${args.secret.api_key}&v=${version}`, {
```
Comment (robinschuster) [inline modules/yext/src/module.ts]: please use the `params` option from axios for submitting query params

File: modules/yext/src/module.ts (hunk 2 above)
Comment (AlexTeusz) [reply modules/yext/src/module.ts]: done

File: modules/yext/src/module.ts (hunk 3 above)
Comment (AlexTeusz) [reply modules/yext/src/module.ts]: done

File: modules/yext/src/module.ts (hunk 4 above)
Comment (AlexTeusz) [reply modules/yext/src/module.ts]: done

File: modules/yext/src/module.ts (hunk 5 above)
Comment (AlexTeusz) [reply modules/yext/src/module.ts]: done

**Comment by AlexTeusz** [comment]
I don't know what goes wrong in your case. I can execute the CreateLocation and GetLocationByFilter nodes without any error.

File: modules/yext/src/module.ts (hunk 1 above)
Comment (AlexTeusz) [reply modules/yext/src/module.ts]: In the README.md it says that you have to query this url to get all **categoryIds**: https://api.yext.com/v2/categories?api_key=<API_KEY>&v=20190424

**Comment by AlexTeusz** [comment]
@robinschuster 
I tested it a view times now and I don't get a 400. I added argument checks for CreateLocation Node to let the user fill all required fields. Please test :)

**Comment by robinschuster** [comment]
I am still getting an error 400 when using getLocationsByFilter.
This node also needs more documentation. I am not sure whether i misconfigured it, because their API reference does not tell me how to pass the filter arguments for that node in our product.

**Comment by AlexTeusz** [comment]
I added detailed documentation to the GetLocationsByFilter Node.

**Comment by AlexTeusz** [comment]
@robinschuster I added a detailed description for the CreateLocation node

**Comment by AlexTeusz** [comment]
Okay we should create one together on monday.

**Pull request merged**